presto-main/src/main/java/io/prestosql/sql/planner/LogicalPlanner.java

```diff
@@ -448,11 +453,16 @@ else if (isUsePreferredWritePartitioning(session)) {
             }
         }
 
-        if (!statisticsMetadata.isEmpty()) {
-            verify(columnNames.size() == symbols.size(), "columnNames.size() != symbols.size(): %s and %s", columnNames, symbols);
-            Map<String, Symbol> columnToSymbolMap = zip(columnNames.stream(), symbols.stream(), SimpleImmutableEntry::new)
-                    .collect(toImmutableMap(Entry::getKey, Entry::getValue));
+        verify(columnNames.size() == symbols.size(), "columnNames.size() != symbols.size(): %s and %s", columnNames, symbols);
+        Map<String, Symbol> columnToSymbolMap = zip(columnNames.stream(), symbols.stream(), SimpleImmutableEntry::new)
+                .collect(toImmutableMap(Entry::getKey, Entry::getValue));
+
+        Set<Symbol> notNullColumnSymbols = columnMetadataList.stream()
+                .filter(column -> !column.isNullable())
+                .map(column -> requireNonNull(columnToSymbolMap.get(column.getName()), "columnToSymbolMap is missing column " + column.getName()))
+                .collect(Collectors.toSet());
```

Comment:
I'd just do this:

```suggestion
                .map(columnToSymbolMap::get)
                .collect(toImmutableSet());
```

It's more concise, and the immutable set will catch any nulls that might result from the lookup (which is a bug somewhere in the implementation, anyway)

Reply:
Indeed, that's much nicer; changed as you suggested by a forced comment.  Thanks very much for looking!

Reply:
It looks like this caused some tests to start failing with `NullPointerException`

Reply:
Oh, yeah, sorry, I missed the following in my suggestion above:

```
.map(ColumnMetadata::getName)
```

Reply:
Doh, I should have looked more closely rather than blindly applying change.  I need to be more careful pushing changes at an hour when I'm no longer sentient ;)

It's fixed now in the latest forced push.